### ISSUE TITLE
Work on improving configuration and running locally

### ISSUE THREAD

**Issue by erezmus** [libero/reviewer#779]
Exploration of how we can run and test the different services more easily. Objectives include
* running the services should "just work" with a make command
* it should be clear where the configuration values are coming from

Things Done:
* use environment variables instead of json config files for system level configuration (e.g. database connection settings
* default environment variables live in docker compose files so that running them only involves running a make command
* look at how publisher article-store is doing things (Makefile, Dockerfile and docker-compose.*.yml) files (https://github.com/libero/article-store)
* make client compiled code completely static

**Comment by erezmus** [comment]
Specifically for the reviewer-client, we are currently loading config from the '/config' endpoint which returns the api url and the login url in an asynchronous fashion should be constants at build time

**Comment by erkannt** [comment]
[Discussion in Infrastructure Office Hours](https://docs.google.com/document/d/1NCZc0J2KC0JBiv6pz-US1EskRMzIPDhptkip4wPzxO0/edit?ts=5e830380#heading=h.nbmi1gxumejv)

tldr:

- infra config as env vars
- product config as conf files
  - default values suitable for local dev/testing
  - override with mount/configmap
- sensitive configs via file, blank/dummy by default for dev/tests
  - as file so that it can be loaded from vault via init container/sidecar [see here](https://www.hashicorp.com/blog/injecting-vault-secrets-into-kubernetes-pods-via-a-sidecar/)

d'accord @erezmus ?

**Comment by erezmus** [comment]
i've now moved config for submission to env variables and same with reviewer-client. Got client is now running in container with multi stage dockerfile inpsired by the one in article store

tomorrow i'll see if i can get them to work smoothly locally.

**Comment by erezmus** [comment]
> d'accord @erezmus ?

lgtm

**Comment by erezmus** [comment]
Progress so far:
- proxying to webpack dev server via separate nginx container after discussing with @erkannt 
- can run client with mocks only for continuum api with reviewer-submission and continuum-adaptor started externally in their own repository

next steps:
- get it working client app + reviewer mocks
- check client browser tests working
- check performance and usability with other team members

**Comment by erezmus** [comment]
There are now three draft PRs for this work:
https://github.com/libero/continuum-adaptor/pull/64
https://github.com/libero/reviewer-submission/pull/164
https://github.com/libero/reviewer-client/pull/211

There are two scenarios to run reviewer locally:
1. Mock continuum only:
This means that you need to run reviewer-submission and continuum-adaptor locally from their own repos

Steps:
- in reviewer-submission repo: `make start`
- in reviewer-submission repo: `make start`

**Comment by erezmus** [comment]
After talking to @de-code, we plan on removing the API_HOST and LOGIN_URL variables in the client so as to make the compiled js completely static.

* The api url is a relative path so that can be removed 
* The continuum url will be made relative (e.g. /continuum-login) and that will be proxied by nginx to the continuum-adaptor service. This will then redirect to the continuum login url. This means that we keep all continuum login configuration in the continuum-adaptor service. The rest of the authentication flow is unchanged

@diversemix @Cooryd @norisalsaadie @giorgiosironi @will-byrne

**Comment by erezmus** [comment]
OK so all 3 PRs are now passing. Still need to implement that change above, but if people could try out on the mac to see performance that would be great

**Comment by will-byrne** [comment]
tested on mac and running `make start_test` seems to fail, see attached.
[docker-output.txt](https://github.com/libero/reviewer/files/4558998/docker-output.txt)
